### ISSUE TITLE
[Bug]: pessimistic transaction and rc mode: deadlock situaction detection failed

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#9909]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):767517b81caf4a05b3c0885eaadb52d30700ce94
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

start mode: ./mo-service -launch ./etc/launch-tae-CN-tae-DN/launch.toml 
deadlock situation :
transaction 1:
<img width="610" alt="企业微信截图_d65d8067-c9c1-482a-ab04-c499fc076b50" src="https://github.com/matrixorigin/matrixone/assets/113406637/6b08831d-3d0a-4d02-97ff-23eac54391f9">

transaction 2:
<img width="738" alt="企业微信截图_35afb591-719f-4c79-a2a1-ad6aee2360d5" src="https://github.com/matrixorigin/matrixone/assets/113406637/f7583037-f9f6-4a98-8649-18da3025d0f4">

deadlock detection failed, ctrl+c can abort
![image](https://github.com/matrixorigin/matrixone/assets/113406637/c63836a2-f302-4909-a716-507cbea6d2f5)

### Steps to Reproduce

```Markdown
create database db;
use db;
create table a(c1 int,c2 int);
insert into a values(1,2),(5,6);
create table b (c1 int,c2 int);
insert into b values(9,0),(7,8);

begin;
update a set c1=99 where c2=2;
-- @session:id=1{
begin;
update b set c1= 88 where c2=0;
-- @session}
update b set c1=55 where c2=0;
-- @session:id=1{
update a set c1=43 where c2=2;
-- @session}
```

**Comment by zhangxu19830126** [comment]
not working on this issue

**Comment by zhangxu19830126** [comment]
fixed by #9976
